scipy/special/add_newdocs.py

```diff
@@ -2693,22 +2693,36 @@ def add_newdoc(name, doc):
 
     Gamma function.
 
+    The Gamma function is defined as
+
     .. math::
 
-          \Gamma(z) = \int_0^\infty x^{z-1} e^{-x} dx = (z - 1)!
+       \Gamma(z) = \int_0^\infty t^{z-1} e^{-t} dt
 
-    The gamma function is often referred to as the generalized
-    factorial since ``z*gamma(z) = gamma(z+1)`` and ``gamma(n+1) =
-    n!`` for natural number *n*.
+    for :math:`\Re(z) > 0` and is extended to the rest of the complex
+    plane by analytic continuation. See [dlmf]_ for more details.
 
     Parameters
     ----------
-    z : float or complex array_like
+    z : array_like
+        Real or complex valued argument
 
     Returns
     -------
-    float or complex
-        The value(s) of gamma(z)
+    scalar or ndarray
+        Values of the Gamma function
+
+    Notes
+    -----
+    The gamma function is often referred to as the generalized
+    factorial since :math:`\Gamma(n + 1) = n!` for natural numbers
+    :math:`n`. It also satisfies the recurrence relation
+    :math:`\Gamma(z + 1) = z \cdot \Gamma(z)`.
```

Comment:
Minor: maybe explicitly state that the recurrence holds for real z, hence for natural values z=n, it becomes m!

Reply:
Updated the language in https://github.com/scipy/scipy/pull/10665/commits/af706ebbb0048fa6d337703935282b4b03f6d143, what do you think?

Reply:
Great